plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,156 @@
+                                    else {
+                                        schemaString.append("\"" + attribute.getName() + "\":" + " {" +
+                                                "\"type\": \"string\"," + "\"enum\": [");
+
+                                        for (Object obj : attribute.type.getEnumConstants()) {
+                                            schemaString.append("\"" + obj + "\",");
+                                        }
+                                         schemaString.append("]},");
+                                    }
+                                } else {
+                                    schemaString.append("\"" + attribute.getName() + "\":");
+                                    switch (attribute.type.getName()){
+
+                                        case "java.lang.String":
+                                            schemaString.append("{\n\"type\": \"string\"},\n");
+                                            break;
+
+                                        case "int":
+                                            schemaString.append("{\n\"type\": \"integer\"},\n");
+                                            break;
+
+                                        case "boolean":
+                                             schemaString.append("{\n\"type\": \"boolean\"},\n");
+                                            break;
+
+                                        case "java.lang.Boolean":
+                                            schemaString.append("{\"type\": \"boolean\"},\n");
+                                            break;
+
+                                        case "java.lang.Integer":
+                                            schemaString.append("{\n\"type\": \"integer\"},\n");
+                                            break;
+
+                                        case "java.lang.Long":
+                                            schemaString.append("{\n\"type\": \"integer\"},\n");
+                                            break;
+
+                                        default:
+                                            schemaString.append("{\n\"type\":  \"object\",\n" +
+                                                    "\"$ref\": \"#/definitions/" + attribute.type.getName() + "\"},\n");
+                                            break;
+                                    }
+
+                                }
+                            }
+                        }
+                    }
+                }
+          }
+
+        /**
+         * Used to generate the schema for the implementors of
+         * the HetroDescribable Configurator
+         * It mimics the HetroDescribable Configurator
+         */
+        ConfigurationAsCode configurationAsCode = ConfigurationAsCode.get();
+        configurationAsCode.getConfigurators()
+                .stream()
+                .forEach(configurator -> {
```

Comment:
You shouldn't have to loop over the object again.
We already have right above.

Reply:
Yeah fixed it +1